tools/walletextension/wallet_extension.go

```diff
@@ -35,14 +50,16 @@ type WalletExtension struct {
 	server            *http.Server
 }
 
-func NewWalletExtension(
-	enclavePrivateKey *ecdsa.PrivateKey,
-	nodeAddr string,
-	viewingKeyChannel chan<- ViewingKey,
-) *WalletExtension {
+func NewWalletExtension(config Config) *WalletExtension {
+	enclavePrivateKey, err := crypto.GenerateKey()
+	if err != nil {
+		panic(err)
+	}
+	viewingKeyChannel := make(chan ViewingKey)
+
 	return &WalletExtension{
-		enclavePrivateKey: enclavePrivateKey,
-		nodeAddr:          nodeAddr,
+		enclavePublicKey:  &enclavePrivateKey.PublicKey,
```

Comment:
Is this actually a public key from the enclave for sending things back to it? I know you're coming back to it immediately probably but if so could maybe stick a todo on the key generation above saying where that will be coming from.

Reply:
I think I'll skip todos as basically everything is todo atm - there's no submission of viewing keys, no enclave pub key, no decryption of results...